src/Form/AdminSettingsForm.php

```diff
@@ -0,0 +1,399 @@
+    return new static(
+      $container->get('entity.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, Request $request = NULL) {
+    $form['credential'] = $this->buildCredentialForm();
+
+    // Data collection settings.
+    $form['data_collection_settings'] = [
+      '#type' => 'vertical_tabs',
+      '#title' => t('Data collection settings'),
+    ];
+    $form['identity'] = $this->buildIdentityForm();
+    $form['field_mappings'] = $this->buildFieldMappingsForm();
+    $form['visibility'] = $this->buildVisibilityForm();
+    $form['thumbnail_url'] = $this->buildThumbnailUrlForm();
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * Build credential form.
+   *
+   * @return array
+   *   Credential form.
+   */
+  private function buildCredentialForm() {
+    $credential_settings = $this->config('acquia_lift.settings')->get('credential');
+
+    $form = [
+      '#title' => t('Credential'),
+      '#type' => 'details',
+      '#tree' => TRUE,
+      '#open' => SettingsHelper::isInvalidCredential($credential_settings),
+    ];
+    $form['account_name'] = [
+      '#type' => 'textfield',
+      '#title' => t('Account Name'),
+      '#default_value' => $credential_settings['account_name'],
+      '#required' => TRUE,
+    ];
+    $form['customer_site'] = [
+      '#type' => 'textfield',
+      '#title' => t('Customer Site'),
+      '#default_value' => $credential_settings['customer_site'],
+    ];
+    $form['api_url'] = [
+      '#type' => 'textfield',
+      '#title' => t('API URL'),
+      '#field_prefix' => 'http(s)://',
+      '#default_value' => $credential_settings['api_url'],
+      '#required' => TRUE,
+    ];
+    $form['access_key'] = [
+      '#type' => 'textfield',
+      '#title' => t('API Access Key'),
+      '#default_value' => $credential_settings['access_key'],
+      '#required' => TRUE,
+    ];
+    $form['secret_key'] = [
+      '#type' => 'password',
+      '#title' => t('API Secret Key'),
+      '#default_value' => $credential_settings['secret_key'],
+      '#required' => empty($credential_settings['secret_key']),
+      '#description' => !empty($credential_settings['secret_key']) ? t('Only necessary if updating') : '',
+    ];
+    $form['js_path'] = [
+      '#type' => 'textfield',
+      '#title' => t('JavaScript Path'),
+      '#field_prefix' => 'http(s)://',
+      '#default_value' => $credential_settings['js_path'],
+      '#required' => TRUE,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Build identity form.
+   *
+   * @return array
+   *   Identity form.
+   */
+  private function buildIdentityForm() {
+    $identity_settings = $this->config('acquia_lift.settings')->get('identity');
+
+    $form = [
+      '#title' => t('Identity'),
+      '#type' => 'details',
+      '#tree' => TRUE,
+      '#group' => 'data_collection_settings',
+    ];
+//    $form['capture_identity'] = [
```

Comment:
Capture identity doesn't work yet. I ran out of time then and the feature was not essential. For clarity, I just commented out the attempted hook at 98285c1499371ecc697b72b2fd4c070588af2e92. We'll revisit this feature's implementation in the future.